### DISCUSSION TITLE
Importer can't import block after block_renaming

### DISCUSSION THREAD

**Discussion by Gashmen** [mozman/ezdxf#808] [Q&A] [answered]
Hello @mozman.

I encountered a peculiarity when Importing a block after renaming in library_doc.
So I have dxf_file_lib like a lib, with all blocks. And when I want to import block that I need importer won't add in doc.blocks.

Previous, I changed names in dxf_file_lib with using function doc.blocks.rename_block(old_name,new_name). 
I changed all names that started with U, and create some structures in block like: if block has inserts, that need to rename it. 
And example I has blocks like : 'Main_name', 'Main_name_U45', 'Main_name_U587', 'Main_name_U587_U659' and ...

On JPG N1 I want to show all names of block_layout. 

![1](https://user-images.githubusercontent.com/90321185/209078514-ce05d98b-1b5f-4b96-b6d2-71fc03804ef5.JPG)

After I used code: 
doc = ezdxf.readfile(dxf_file_lib)
doc_new_file = ezdxf.new(dxfversion='AC1024')
importer = Importer(doc, doc_new_file)
list_with_block_names = [block.dxf.name for block in doc.blocks if '*' not in block.dxf.name]
for i in list_with_block_names:
    importer.import_block(i)
importer.finalize()

And then:

![2](https://user-images.githubusercontent.com/90321185/209079309-8dfbf74b-d434-431a-8988-e613b6361224.JPG)

I was checked the dxf_file_lib in Notepad and I saw, that they have names U22 and U22 is AcDbBlockReference or Insert
![3](https://user-images.githubusercontent.com/90321185/209079669-700335a7-5856-44c9-81f4-634155954001.JPG)

What I need to do for fix it? Easy way is to change it  with open dxf like txt file and search U_blocks, but mb you have a solution.
Thank you!

**Comment by mozman** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/4230966/209095049-8fffc4e6-50e6-4dde-b09f-b5dcd1fa8984.png)

**Reply by Gashmen** [reply]
Thank you! It's very sad :(
